development-guild/rewards/generate-weekly-csvs.js

```diff
@@ -0,0 +1,131 @@
+const fs = require('fs')
+const Papa = require('papaparse')
+const { stringify } = require('csv-stringify');
+
+generate()
+
+async function generate() {
+  console.log('generate')
+
+  // Load the Coordinape contributions
+  const contributionsFilePath = `../contributions/coordinape-contributions.csv`
+  console.debug('contributionsFilePath:', contributionsFilePath)
+  let contributionsData = []
+  if (!fs.existsSync(contributionsFilePath)) {
+    console.error('File does not exist')
+    return
+  } else {
+    contributionsData = await loadCSVData(contributionsFilePath)
+  }
+  // console.debug('contributionsData:', contributionsData)
+
+  // Iterate every week from the week of [Sun May-29-2022 → Sun Jun-05-2022] until now
+  const weekEndDate = new Date('2022-06-05T00:00:00Z')
+  console.debug('weekEndDate:', weekEndDate)
+  const nowDate = new Date()
+  console.debug('nowDate:', nowDate)
+  while (nowDate.getTime() > weekEndDate.getTime()) {
+    const weekBeginDate = new Date(weekEndDate.getTime() - 7*24*60*60*1000)
+    console.info('week:', `[${weekBeginDate.toISOString()} → ${weekEndDate.toISOString()}]`)
+
+    // Extract this week's contributions
+    let contributionsDataThisWeek = []
+    contributionsData.forEach((dataRow) => {
+      const updatedAt = dataRow.updated_at
+      // console.debug('updatedAt:', updatedAt)
+      if (updatedAt != undefined) {
+        const updatedAtDate = new Date(updatedAt)
+        // console.debug('updatedAtDate:', updatedAtDate)
+        if (updatedAtDate.getTime() >= weekBeginDate.getTime()
+            && updatedAtDate.getTime() < weekEndDate.getTime()
+        ) {
+          console.debug('updatedAtDate:', updatedAtDate)
+          contributionsDataThisWeek.push(dataRow)
+        }
+      }
+    })
+    console.debug('contributionsDataThisWeek:', contributionsDataThisWeek)
+
+    // Summarize hours spent for each citizen
+    let rewards = {}
+    contributionsDataThisWeek.forEach((dataRow) => {
+      if (rewards[String(dataRow.passport_id)] == undefined) {
+        const reward = {
+          passport_id: Number(dataRow.passport_id),
+          profile_id: Number(dataRow.profile_id),
+          hours_spent_total: Number(dataRow.hours_spent)
+        }
+        rewards[String(dataRow.passport_id)] = reward
+      } else {
+        const reward = rewards[String(dataRow.passport_id)]
+        reward.hours_spent_total += Number(dataRow.hours_spent)
+        reward.hours_spent_total = Number(Number(reward.hours_spent_total).toFixed(2))
```

Comment:
This line is quite nuts, do you really have to do that? 

`let hours =  +dataRow.hours_spent + dataRow.hours_spent_total
hours = hours.toFixed(2)
reward.hours_spent_total = hours`

I dunnot maybe not worth it, jsut talk me  while to read!

Reply:
> This line is quite nuts, do you really have to do that?
> 
> `let hours = +dataRow.hours_spent + dataRow.hours_spent_total hours = hours.toFixed(2) reward.hours_spent_total = hours`
> 
> I dunnot maybe not worth it, jsut talk me while to read!

Hehe, I agree, nuts indeed 🤪

The reason for having it was just to store numbers as 1.95 instead of 1.95000000001.  But I hope that there are more elegant solutions for solving this.